src/hooks/useCreateGrant.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback } from 'react';
+import { useSigner, useAccount } from 'wagmi';
+
+import { functionRequest } from '../supabase';
+
+const domain = {
+  name: 'ENS Grants',
+  version: '1',
+  chainId: 1,
+};
+
+const types = {
+  Grant: [
+    { name: 'address', type: 'address' },
+    { name: 'roundId', type: 'uint256' },
+    { name: 'title', type: 'string' },
+    { name: 'description', type: 'string' },
+    { name: 'fullText', type: 'string' },
+  ],
+};
+
+export function useCreateGrant() {
```

Comment:
Maybe thinking ahead too much, but will users be able to edit grants after they are created? Or not worrying about that for now?

Reply:
Not worrying for now, but when we add that in the UI, the way it would actually work is creating a new grant, which automatically replaces the old one. So hook-wise, we use this hook either way.